mat64/dense.go

```diff
@@ -65,6 +65,11 @@ type Dense struct {
 	capRows, capCols int
 }
 
+// NewDense creates a new matrix of type *Dense with dimensions r and c. If the mat argument is nil,
+// a new data slice is allocated.
+// Data slice is of Row-major order, i.e., the (i, j) element in matrix should be at (i*c + j)-th
+// position in mat.
+// Also note that NewDense(0, 0, nil) can be used for undetermined size matrix initialization.
```

Comment:
This comment is better handled in an example. The behavior is already documented in the package-level documentation, and it's misleading because it makes it seem like those arguments are a special case even though they aren't. NewDense(0, 0, mat) can also be used for undetermined size.